python/src/scipp/io/hdf5.py

```diff
@@ -108,24 +228,50 @@ def _read_data_array(self, group):
                 c = category
                 if category == 'coords' and not g.attrs.get('aligned', True):
                     c = 'unaligned_coords'
-                contents[c][name] = self._read_variable(g)
+                contents[c][name] = VariableIO.read(g)
         return sc.DataArray(**contents)
 
 
-def data_array_to_hdf5(self, filename):
+class DatasetIO:
+    @staticmethod
+    def write(group, data):
+        _write_scipp_header(group, 'Dataset')
+        # Slight redundancy here from writing aligned coords for each item,
+        # but irrelevant for common case of 1D coords with 2D (or higher)
+        # data. The advantage is the we can read individual dataset entries
+        # directly as data arrays.
+        for name in data:
+            HDF5IO.write(group.create_group(name), data[name])
+
+    @staticmethod
+    def read(group):
+        _check_scipp_header(group, 'Dataset')
+        from .._scipp import core as sc
+        return sc.Dataset({name: HDF5IO.read(group[name]) for name in group})
+
+
+class HDF5IO:
+    _handlers = dict(
```

Comment:
This is nice and extendable. I like the way this is put together.

Reply:
You are encoding version information that allow us to potentially deal with future changes to layout or extension of data structure. How would you see that being done? maybe versioned handler dict or something similar packing the right `classIO` classes?

Reply:
Yes, something like that. I hope we will not have to do that once we havsettled on a schema (given that our overall data structure is changing little, and we would mainly deal with dtype *additions*), but wanted to have the version info there for future proofing.